ext/include/opentelemetry/ext/zpages/static/tracez_index.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+const char tracez_index[] =
```

Comment:
I think it would have been a bit cleaner if we could get the contents from a filesystem for that. Even if we are not supposed to host the HTML + JS in this repo, I think embedding it in C source code is a bit odd.. I don't think it's a blocker, but we should refactor this to load the resources at runtime from a filesystem rather than hardcoding. For perf reasons you can (optionally) read the HTML and JS files contents into char array once at server start.

Reply:
Yeah, I agree I prefer not embedding the static files as C. One reason we did this was because OT Java also does something similar by emitting their HTML using strings. The main reason we did this (over serving files) was that it's the easiest way to serve files without either adding workarounds to get the executable to serve files from the right filesystem path (also unsure how this would work?) or complicate the instructions to run zPages (i.e. users have to go to the static file directory and run executable from there).

Maybe this is easier than we think and you have some insight on the best course of action? Otherwise, we can address this later and I open a Github issue?

Reply:
Makes sense. I think it would be good to enable both modes:
- serving from known location relative to the main executable
- embedding the HTML / JS within

Basically a combo approach that allows to ship it either this or that way. Some extra info (for future) on how we could have used some form of config option to decide: https://stackoverflow.com/questions/410980/include-a-text-file-in-a-c-program-as-a-char - see the 2nd answer for C++.

If embedding - then perform an extra step in CMake that transforms the .html / .js / .css files for inclusion using raw string literals. Something that takes `index.html` and creates `index.html.h` out of it (generated at build time). Including it using above method.

If not embedding, then dynamically reading the file contents into array at runtime.

Not for this PR, but something we can consider for future.

Reply:
For CMake, would something similar be possible for Bazel? I haven't heard about build functionality like that before, and it sounds cool.

Anyways, that looks like the ideal scenario with lots of useful information and I'll open a Github issue for it once this PR is merged!

Reply:
It is fairly common to generate some source and/or header files at build time, then compile them into the main executable. Most build systems offer a few ways to make that happen. I know how to do this for CMake and MSBuild, I am sure Bazel has something like this. The other alternative is to have an outside shell script that prepares the files for you ahead of time, then your build script either includes them or not, depending on whether you embed or serve from a filesystem.